File: Day159_2025/Q159_Leetcode.java
import java.util.*;
class Q159_Leetcode {
    public List<Integer> lexicalOrder(int n) {
        List<Integer> ans = new ArrayList<>();
        int x = 1;
        for (int i = 0; i < n; i++) {
            ans.add(x);
            if (x * 10 <= n) {
                x *= 10;
            } else {
                if (x >= n) x /= 10;
                x++;
                while (x % 10 == 0) x /= 10;
            }
        }
        return ans;
    }
}

File: Day159_2025/Q159_Leetcode.py
from typing import List


class Solution:
    def lexicalOrder(self, n: int) -> List[int]:
        ans = []
        x = 1
        for _ in range(n):
            ans.append(x)
            if x * 10 <= n:
                x *= 10
            else:
                if x >= n:
                    x //= 10
                x += 1
                while x % 10 == 0:
                    x //= 10
        return ans
